### ISSUE TITLE
Issue updating state

### ISSUE THREAD

**Issue by costular** [uniflow-kt/uniflow-kt#55]
Hey!

I've found an issue that state don't get updated when you try to update the state from a flow `onEach` method which have another flow concatenated afterwards. Or even crashing when you merge (using `flatmapMerge` or `flatmapLatest` - which run in parallel).

```
java.lang.IllegalStateException: Flow invariant is violated:
		Emission from another coroutine is detected.
		Child of "coroutine#6":ProducerCoroutine{Active}@82c57b3, expected child of "coroutine#1":UndispatchedCoroutine{Active}@5be82d43.
		FlowCollector is not thread-safe and concurrent emissions are prohibited.
		To mitigate this restriction please use 'channelFlow' builder instead of 'flow'
```

To be honest I haven't checked the internal code and don't know which are your expectations. However, is a really bad experience not being able to update the state when executing two flows or more

Thanks!

**Comment by arnaudgiuliani** [comment]
yes it should be fixed with new engine in `0.13.x`. Uniflow was relying on `Flow` as internal plumbing. It has been kicked out for simpler design.

**Comment by arnaudgiuliani** [comment]
Please check latest version of uniflow: https://search.maven.org/search?q=org.uniflow-kt
